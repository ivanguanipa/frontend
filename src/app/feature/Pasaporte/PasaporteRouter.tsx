import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { LazyFallback } from 'app/shared/components/LazyFallback';

const MainPage = React.lazy(() => import('./pages/Main'));
const ShowPage = React.lazy(() => import('./pages/Show'));

export const PasaporteRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    {/* Layout compartido entre las rutas va aquÃ­ */}
    <Switch>
      <Route path="/pasaportes" exact component={MainPage}></Route>
      <Route path="/pasaportes/show/:id" component={ShowPage}></Route>

    </Switch>
  </React.Suspense>
);
