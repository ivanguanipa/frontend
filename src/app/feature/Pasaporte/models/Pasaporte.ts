type Nullable<T> = T | null ;
export interface Pasaporte {
    id?: Nullable<number>,
    fullname: string,
    address: string,
    birthdate: string,
    documentId: number,
    applicationDate: string,
    appointmentDate?: Nullable<string>,
    amount?: Nullable<number>,
}