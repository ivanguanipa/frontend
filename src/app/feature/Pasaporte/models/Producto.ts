export interface Producto {
  title: string;
  slug: string;
  body: string;
  createdAt: string;
  updatedAt: string;
}
