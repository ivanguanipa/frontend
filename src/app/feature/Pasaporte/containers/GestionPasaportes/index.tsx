import * as PropTypes from 'prop-types';
import * as React from 'react';
import { DivContainer, DivRow } from './styles';
import { FormCrearPasaporte } from '../../components/FormCrearPasaporte';
import { ListaPasaportes } from '../../components/ListarPasaportes';
import { PaginadorPasaportes } from '../../components/PaginadorPasaportes';
import { Pasaporte } from '../../models/Pasaporte';
import { useEffect } from 'react';

interface GestionPasaportesProps {
  pasaportes: Array<Pasaporte>;
  listarPasaportes: (numeroPagina: number) => void;
  agregarNuevoPasaporte: (pasaportes: Pasaporte) => void;
  eliminarPasaporte: (pasaportes: Pasaporte) => void;
  cantidadTotalPasaporte: number;
}

export const GestionPasaportes: React.FC<GestionPasaportesProps> = ({
  agregarNuevoPasaporte,
  pasaportes,
  listarPasaportes,
  eliminarPasaporte,
  cantidadTotalPasaporte,
}) => {
  useEffect(() => {
    console.log('pasando useeffect');
    listarPasaportes(0);
  }, [listarPasaportes]);
  return (
    <DivContainer>
      <DivRow>
        <FormCrearPasaporte
          onSubmit={agregarNuevoPasaporte}
          formTitle="Crear pasaporte"
        />
      </DivRow>
      <DivRow>
        <ListaPasaportes
          pasaportes={pasaportes}
          onClickEliminarPasaporte={eliminarPasaporte}
        />
        <PaginadorPasaportes
          cantidadTotalPasaportes={cantidadTotalPasaporte}
          onClickCambiarPagina={listarPasaportes}
        />
      </DivRow>
    </DivContainer>
  );
};

GestionPasaportes.propTypes = {
  pasaportes: PropTypes.array.isRequired,
  listarPasaportes: PropTypes.func.isRequired,
  agregarNuevoPasaporte: PropTypes.func.isRequired,
  eliminarPasaporte: PropTypes.func.isRequired,
  cantidadTotalPasaporte: PropTypes.number.isRequired,
};
