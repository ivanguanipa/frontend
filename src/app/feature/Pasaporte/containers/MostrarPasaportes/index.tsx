import * as PropTypes from 'prop-types';
import * as React from 'react';
import { DivContainer, DivRow3 } from './styles';
import { Pasaporte } from '../../models/Pasaporte';
import { useEffect } from 'react';
import { Button } from 'app/shared/components/Button';
import { useParams } from "react-router-dom";
import { useHistory } from "react-router-dom";
import {dateFormat} from "app/utils/index";

interface MostrarPasaportesProps {
  mostrarPasaporte: (id: string, callback:any) => void;
  pasaporte?: Pasaporte;
}
export const MostrarPasaportes: React.FC<MostrarPasaportesProps> = ({
  mostrarPasaporte,
  pasaporte,
}) => {
  const { id } = useParams<{ id: string }>();

  const pleaseWait = 'Por favor espere...';
  let history = useHistory();
  useEffect(() => {
    const recordNotFound = ()=>{
      alert('no encontrado');
      history.push("/pasaportes");
    }
    mostrarPasaporte(id, recordNotFound);
  }, [id,mostrarPasaporte,history]);

  const muestraPasaporte = (data:Pasaporte)=>{
      return  <div>
      <div>Nombre y Apellido: {data.fullname}</div>
      <div>Cédula: {data.documentId}</div>
      <div>Dirección: {data.address}</div>
      <div>Fecha de Nacimiento: {dateFormat(data.birthdate,"DD-MM-YYYY")} </div>
      <div>Fecha de la solicitud: {dateFormat(data.applicationDate,"DD-MM-YYYY")}</div>
      <div>Fecha de la cita: {dateFormat(String(data.appointmentDate),"DD-MM-YYYY")}</div>
      <div>Monto a cancelar: {data.amount}</div>
      </div>
  }
  return (
    <DivContainer>
      <DivRow3>
        <Button onClick={() => { history.push('/pasaportes') }}>
          <span role="img" aria-labelledby="trash">
            Regresar
          </span>
        </Button>
        {
          !pasaporte ? pleaseWait : muestraPasaporte(pasaporte)
        }
      </DivRow3>
    </DivContainer>
  );
};

MostrarPasaportes.propTypes = {
  pasaporte: PropTypes.shape({
    fullname: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    birthdate: PropTypes.string.isRequired,
    documentId: PropTypes.number.isRequired,
    applicationDate: PropTypes.string.isRequired,
  }).isRequired,
  mostrarPasaporte: PropTypes.func.isRequired,
};
