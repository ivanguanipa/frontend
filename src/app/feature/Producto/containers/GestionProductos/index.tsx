import * as PropTypes from 'prop-types';
import * as React from 'react';
import { DivContainer, DivRow } from './styles';
import { FormCrearProducto } from '../../components/FormCrearProducto';
import { ListaProductos } from '../../components/ListarProductos';
import { PaginadorProductos } from '../../components/PaginadorProductos';
import { Producto } from '../../models/Producto';
import { useEffect } from 'react';

interface GestionProductosProps {
  productos: Array<Producto>;
  listarProductos: (numeroPagina: number) => void;
  agregarNuevoProducto: (productos: Producto) => void;
  eliminarProducto: (productos: Producto) => void;
  cantidadTotalProducto: number;
}

export const GestionProductos: React.FC<GestionProductosProps> = ({
  agregarNuevoProducto,
  productos,
  listarProductos,
  eliminarProducto,
  cantidadTotalProducto,
}) => {
  useEffect(() => {
    listarProductos(0);
  }, [listarProductos]);
  return (
    <DivContainer>
      <DivRow>
        <FormCrearProducto
          onSubmit={agregarNuevoProducto}
          formTitle="Crear producto"
        />
      </DivRow>
      <DivRow>
        <ListaProductos
          productos={productos}
          onClickEliminarProducto={eliminarProducto}
        />
        <PaginadorProductos
          cantidadTotalProductos={cantidadTotalProducto}
          onClickCambiarPagina={listarProductos}
        />
      </DivRow>
    </DivContainer>
  );
};

GestionProductos.propTypes = {
  productos: PropTypes.array.isRequired,
  listarProductos: PropTypes.func.isRequired,
  agregarNuevoProducto: PropTypes.func.isRequired,
  eliminarProducto: PropTypes.func.isRequired,
  cantidadTotalProducto: PropTypes.number.isRequired,
};
