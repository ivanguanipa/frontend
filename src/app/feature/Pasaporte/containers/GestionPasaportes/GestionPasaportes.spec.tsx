import * as React from 'react';
import { ShallowWrapper, shallow, render, mount } from 'enzyme';
import { GestionPasaportes } from 'app/feature/Pasaporte/containers/GestionPasaportes';
import { Pasaporte } from 'app/feature/Pasaporte/models/Pasaporte';
import { useEffect } from 'react';


describe('Gestion Pasaporte Test', () => {
  let componentWrapper: ShallowWrapper;
  const pasaportes: Pasaporte[] = [{
    fullname: 'lorem',
    address: 'ipsun',
    birthdate: '2021-01-01',
    documentId: 12345,
    applicationDate: '2021-01-01',
  }];

  afterEach(() => {
    componentWrapper.unmount();
  });

  it('should match snapshot', () => {
    componentWrapper = shallow(
        <GestionPasaportes 
        cantidadTotalPasaporte={1}
        pasaportes={pasaportes} 
        listarPasaportes={()=>{}} 
        agregarNuevoPasaporte={()=>{}} 
        eliminarPasaporte={()=>{}} 
      />
      
    );
    expect(componentWrapper).toMatchSnapshot();
  });
 
  it('deberia testear useeffect', () => {
    const mockListarPasaporte = jest.fn();
    const useEffectaa = jest.spyOn(React, "useEffect").mockImplementation(f => f());
     mount(
        <GestionPasaportes 
        cantidadTotalPasaporte={1}
        pasaportes={pasaportes} 
        listarPasaportes={mockListarPasaporte} 
        agregarNuevoPasaporte={()=>{}} 
        eliminarPasaporte={()=>{}} 
      />
      
    );
    expect(useEffectaa).toHaveBeenCalled();
  });
});

export {};
