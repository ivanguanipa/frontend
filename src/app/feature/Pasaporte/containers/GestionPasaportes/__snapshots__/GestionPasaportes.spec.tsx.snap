// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gestion Pasaporte Test should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <FormCrearPasaporte
      formTitle="Crear pasaporte"
      onSubmit={[Function]}
    />
  </styled.div>
  <styled.div>
    <ListaPasaportes
      onClickEliminarPasaporte={[Function]}
      pasaportes={
        Array [
          Object {
            "address": "ipsun",
            "applicationDate": "2021-01-01",
            "birthdate": "2021-01-01",
            "documentId": 12345,
            "fullname": "lorem",
          },
        ]
      }
    />
    <PaginadorPasaportes
      cantidadTotalPasaportes={1}
      onClickCambiarPagina={[Function]}
    />
  </styled.div>
</styled.div>
`;
